src/Features/Core/Portable/SplitOrMergeIfStatements/Nested/AbstractMergeNestedIfStatementsCodeRefactoringProvider.cs

```diff
@@ -91,9 +91,8 @@ protected sealed override SyntaxNode GetChangedRoot(Document document, SyntaxNod
             // Check whether the statement is a first statement inside an if or else if.
             // If it's inside a block, it has to be the first statement of the block.
 
-            // A statement should always be in a statement container, but we'll do a defensive check anyway so that
-            // we don't crash if the helper is missing some cases or there's a new language feature it didn't account for.
-            Debug.Assert(syntaxFacts.IsStatementContainer(statement.Parent));
+            // We can't assume that a statement will always be in a statement container, because an if statement
+            // in top level code will be in a GlobalStatement.
             if (syntaxFacts.IsStatementContainer(statement.Parent))
```

Comment:
looks like we can't merge nested if's at the top level?  can you check that?  if we have that issue, can you fix or open issue on it?

Reply:
The code already worked for that scenario, but this Debug.Assert was blocking it, so this PR effectively enables it. The test I added validates it. There are existing tests for merging from nested ifs with outer ifs at the top level [here](https://github.com/dotnet/roslyn/blob/main/src/EditorFeatures/CSharpTest/SplitOrMergeIfStatements/MergeNestedIfStatementsTests_WithOuter.cs#L2194)